### PR TITLE
MGDOBR-848: Schema catalog enhancements

### PR DESCRIPTION
**Pull request by jomarko**
This PR is a follow up work for https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/79

The backend processor configuration schemas are crucial for UI forms generation. During review https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/79#pullrequestreview-1016493932 were identiefied backend configurations, that can be slightly updated, to obtain easier to use/read UI.

This PR address especially point 2) and 5) of review https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/79#pullrequestreview-1016493932.

[MGDOBR-848](https://issues.redhat.com/browse/MGDOBR-848)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by r00ta — APPROVED** [review]

**Comment by jomarko** [comment]
@kelvah would you mind to review this PR and eventually approve?

**Review by kelvah — APPROVED** [review]
LGTM

File: processor/configuration/src/main/resources/schemas/sources/aws_sqs_source_0.1.json [hunk 1]
```diff
@@ -111,7 +111,10 @@
       "title": "Protocol",
       "description": "The underlying protocol used to communicate with SQS",
       "type": "string",
-      "example": "http or https",
+      "enum": [
+        "http",
+        "https"
+      ],
```
Comment (danielezonca) [inline processor/configuration/src/main/resources/schemas/sources/aws_sqs_source_0.1.json:117]: @jomarko 
For the record, this file (like all the other RHOC based sinks/sources) is "produced" (manually) from RHOC catalog so we should not change it and ask them to update if necessary. For example this file is derived from [this other](https://github.com/bf2fc6cc711aee1a0c2a/cos-fleet-catalog-camel/blob/main/etc/kubernetes/manifests/base/connectors/connector-catalog-camel-aws/aws_sqs_sink_0.1.json#L99-L105)

Comment (jomarko) [reply processor/configuration/src/main/resources/schemas/sources/aws_sqs_source_0.1.json:117]: @danielezonca ok, so what is your point please? Should I revert the changes? They are quite usefull for RHOSE UI.

Comment (manstis) [reply processor/configuration/src/main/resources/schemas/sources/aws_sqs_source_0.1.json:117]: To be fair @danielezonca if you look at the history of the files they have already had major modification by RHOSE to support features desired in the UI. The JSON schema's may have originated from RHOC but they are very different in RHOSE. See https://github.com/jomarko/sandbox/commit/8c0aef5d5bbbdc07440c382024bfd3bfe3fb98f3 and https://github.com/jomarko/sandbox/commit/8e0b58e985f1c3c14d08ed05460597d29530c328 for examples.

Comment (danielezonca) [reply processor/configuration/src/main/resources/schemas/sources/aws_sqs_source_0.1.json:117]: @manstis 
Yes let's agree with RHOC team how to proceed for similar changes 👍 
Btw the first PR is not touching the `something_sink_0.1` files so it is safe, only the second one is changing the validation part (with regexpr)

**Pull request merged**